misc/userscripts/qute-pass

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+
+# Insert login information using pass and a dmenu-provider (e.g. dmenu, rofi -dmenu, ...).
+# A short demonstration can be seen here: https://i.imgur.com/KN3XuZP.gif.
+#
+# The domain of the site has to appear as a segment in the pass path, for example: "github.com/cryzed" or
+# "websites/github.com". How the username and password are determined is freely configurable using the CLI arguments.
+# The login information is inserted by emulating key events using qutebrowser's fake-key command in this manner:
+# [USERNAME]<Tab>[PASSWORD], which is compatible with almost all login forms.
+#
+# Dependencies: tldextract (Python 3 module), pass
+# For issues and feedback please use: https://github.com/cryzed/qutebrowser-userscripts
+#
+# Chris Braun (cryzed), 2017
+
+import argparse
+import enum
+import fnmatch
+import functools
+import os
+import re
+import shlex
+import subprocess
+import sys
+
+import tldextract
+
+PASSWORD_STORE_PATH = os.path.expanduser('~/.password-store')
+DMENU_INVOCATION = 'rofi -dmenu'
+ENCODING = 'UTF-8'
```

Comment:
Those are all customizable via (commandline) arguments, but then still passed as (Python) argument to the respective functions, despite them being always provided when those functions are called. Wouldn't it make more sense to make those arguments mandatory and only put those values as default for argparse?

Reply:
All except the encoding at least, which I doubt will ever change. Regarding your question: I tried to keep my code as reusable as possible, so if someone wanted to import functions from qute-pass, such as `find_pass_candidates` and `dmenu` they could just do that, and have default sane keyword-arguments. These few are constants so they can be accessed from the `add_argument` calls and inside the function headers.

However, you might be right, the chances are slim that someone will import qute-pass, I'll remove everything except encoding.

Reply:
I need a tip regarding encoding after all, does this actually need to be a CLI switch for different systems? Is it possible that with a fixed UTF-8 encoding this will fail on some systems?